interfaces/python/CMakeLists.txt

```diff
@@ -7,284 +7,272 @@
-                # m6Ot jBCQAJ
-                set_target_properties(
-                    _helics
-                    PROPERTIES LINK_FLAGS "-undefined dynamic_lookup"
-                )
-            else()
-                swig_link_libraries(helics ${Python3_LIBRARIES})
-            endif()
+    include(GNUInstallDirs)
 
-            target_include_directories(_helics PRIVATE ${Python3_INCLUDE_DIRS})
-            if(MSVC)
-                target_compile_options(_helics PRIVATE "/wd4100")
-            else(MSVC)
-                target_compile_options(_helics PRIVATE "-Wno-long-long")
-            endif(MSVC)
-
-            set_target_properties(_helics PROPERTIES FOLDER interfaces)
-
-            install(
-                TARGETS
-                    _helics
-                    DESTINATION
-                    python
-                    COMPONENT
-                    python
-            )
-
-            copy_key_files_to_target_location(_helics)
-            copy_shared_target(_helics helicsSharedLib)
-
-        else()
-            message("Building without swig cmake ${CMAKE_VERSION}")
-            python3_add_library(helicsPYTHON MODULE helicsPython.c)
-
-            copy_key_files_to_target_location(helicsPYTHON PUBLIC)
-            copy_shared_target(helicsPYTHON helicsSharedLib)
-
-            # Set the output library name to _helics
-            set_target_properties(
-                helicsPYTHON
-                PROPERTIES PREFIX "_" OUTPUT_NAME "helics"
-            )
-            target_link_libraries(helicsPYTHON PUBLIC helicsSharedLib)
-
-            set_target_properties(helicsPYTHON PROPERTIES FOLDER interfaces)
-            install(
-                TARGETS
-                    helicsPYTHON
-                    DESTINATION
-                    python
-                    COMPONENT
-                    python
-            )
-
-            configure_file(${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in
-                           ${CMAKE_CURRENT_BINARY_DIR}/setup.py)
-            configure_file(${CMAKE_CURRENT_SOURCE_DIR}/helics.py
-                           ${CMAKE_CURRENT_BINARY_DIR}/helics.py COPYONLY)
+    # Setup rpath
+    set(CMAKE_SKIP_BUILD_RPATH FALSE)
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+    set(CMAKE_MACOSX_RPATH ON)
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
-        endif()
+    # Add the local directory to the rpath
+    if(NOT APPLE)
+        list(APPEND CMAKE_INSTALL_RPATH $ORIGIN)
+        list(APPEND CMAKE_INSTALL_RPATH
+                    "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}"
+        )
     else()
+        list(APPEND CMAKE_INSTALL_RPATH "@loader_path/.")
+        list(APPEND CMAKE_INSTALL_RPATH "@executable_path/.")
+    endif()
+endif()
+
+###############################
+# Find required dependencies
+###############################
+# Find HELICS
+if(NOT hasParent OR NOT TARGET HELICS::helicsSharedLib)
+    find_package(HELICS 2 REQUIRED)
```

Comment:
Is there a minimum version of HELICS this will work with.  like 2.3 or something like that?  Is it going to work with 2.0 for example.

Reply:
2.2.2 is the first version that bundled the helics.i file, but in theory it could work with 2.0 if they provided a path to a valid helics.i file in HELICS_SWIG_helics.i_FILE and set HELICS_SWIG_INCLUDE_DIRS to the right include paths.

Reply:
Does it make sense to leave it at 2 or should we set it to something like 2.3?

Reply:
I'd rather leave it more open to start -- though hoping there aren't any users still on 2.0 might be optimistic.